react/features/security/components/security-dialog/native/SecurityDialog.js

```diff
@@ -0,0 +1,408 @@
+
+        if (!_lobbyModeSwitchVisible) {
+            return null;
+        }
+
+        return (
+            <View>
+                <Text style = { _securityDialogStyles.title } >
+                    { t('lobby.dialogTitle') }
+                </Text>
+                <Text style = { _securityDialogStyles.text } >
+                    { t('lobby.enableDialogText') }
+                </Text>
+                <LobbyModeSwitch
+                    lobbyEnabled = { _lobbyEnabled }
+                    onToggleLobbyMode = { this._onToggleLobbyMode } />
+            </View>
+        );
+    }
+
+    /**
+     * Renders room lock.
+     *
+     * @returns {ReactElement}
+     * @private
+     */
+    _renderRoomLock() {
+        const {
+            _locked,
+            _roomLockSwitchVisible,
+            _securityDialogStyles,
+            t
+        } = this.props;
+        const { showElement } = this.state;
+
+        if (!_roomLockSwitchVisible) {
+            return null;
+        }
+
+        return (
+            <View>
+                <Text style = { _securityDialogStyles.title } >
+                    { t('dialog.lockRoom') }
+                </Text>
+                <Text style = { _securityDialogStyles.text } >
+                    { t('security.about') }
+                </Text>
+                <RoomLockSwitch
+                    locked = { _locked === LOCKED_REMOTELY }
+                    onToggleRoomLock = { this._onToggleRoomLock }
+                    toggleRoomLock = { showElement } />
+                { this._renderRoomLockMessage() }
+            </View>
+        );
+    }
+
+    /**
+     * Renders room lock text input/message.
+     *
+     * @returns {ReactElement}
+     * @private
+     */
+    _renderRoomLockMessage() {
+        let textInputProps = _TEXT_INPUT_PROPS;
+        const {
+            _dialogStyles,
+            _locked,
+            _password,
+            _passwordNumberOfDigits,
+            t
+        } = this.props;
+        const { passwordInputValue, showElement } = this.state;
+
+        if (_passwordNumberOfDigits) {
+            textInputProps = {
+                ...textInputProps,
+                keyboardType: 'numeric',
+                maxLength: _passwordNumberOfDigits
+            };
+        }
+        if (showElement) {
+            if (!_locked) {
+                return (
+                    <TextInput
+                        onChangeText = { this._onChangeText }
+                        placeholder = { t('lobby.passwordField') }
+                        style = { _dialogStyles.field }
+                        underlineColorAndroid = { FIELD_UNDERLINE }
+                        value = { passwordInputValue }
+                        { ...textInputProps } />
+                );
+            } else if (_locked) {
+                if (_locked === LOCKED_LOCALLY) {
+                    return (
+                        <TextInput
+                            editable = { false }
```

Comment:
Please make it editable, the user may want to change the password and it's nice to be able to do it without turning it off and on again.